src/controllers/update/update.review.beneficial.owner.individual.ts

```diff
@@ -55,21 +55,32 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     const boiIndex = req.query.index;
     const appData = getApplicationData(req.session);
 
-    if (boiIndex !== undefined && appData.beneficial_owners_individual && appData.beneficial_owners_individual[Number(boiIndex)].id === req.body["id"]){
-      const boId = appData.beneficial_owners_individual[Number(boiIndex)].id;
-      const dob = appData.beneficial_owners_individual[Number(boiIndex)].date_of_birth as InputDate;
-      const haveDayOfBirth = appData.beneficial_owners_individual[Number(boiIndex)].have_day_of_birth;
+    if (boiIndex !== undefined && appData.beneficial_owners_individual && appData.beneficial_owners_individual[Number(boiIndex)].id === req.body["id"]) {
+      const boData: BeneficialOwnerIndividual = appData.beneficial_owners_individual[Number(boiIndex)];
+      const boId = boData.id;
+      const dob = boData.date_of_birth as InputDate;
+      const haveDayOfBirth = boData.have_day_of_birth;
+
+      let trustIds: string[] = [];
+      if (boData.trust_ids) {
+        trustIds = boData.trust_ids;
```

Comment:
Use the spread operator to recreate a new array or else they will have the same reference [...boData.trust_ids]

Your if condition doesn't check  if the array has a size if (boData?.trust_ids.length);

You can do everything in one line:
const trustIds: string[] = boData?.trust_ids.length ? [...boData.trust_ids] : [];

Reply:
Thanks. Have implemented this now.